content/oci/oci.go

```diff
@@ -398,6 +402,49 @@ func (s *Store) writeIndexFile() error {
 	return os.WriteFile(s.indexPath, indexJSON, 0666)
 }
 
+// GC removes garbage from Store. The garbage to be cleaned are:
+//   - unreferenced (dangling) blobs in Store which have no predecessors
+//   - garbage blobs in the storage whose metadata is not stored in Store
+func (s *Store) GC(ctx context.Context) error {
+	s.sync.Lock()
+	defer s.sync.Unlock()
+
+	// clean up dangling layers in Store
+	danglings := s.graph.GetDanglingLayers()
+	for _, desc := range danglings {
+		// do not remove existing manifests in the index
+		if _, err := s.tagResolver.Resolve(ctx, string(desc.Digest)); err == errdef.ErrNotFound {
+			// remove the blob and its metadata from the Store
+			if err := s.deleteNode(ctx, desc); err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
It seems it only covers the root nodes of the dangling graph.

Think about this senario:
```mermaid
graph TD
  A --> B
  A --> C
  C --> D
  D --> E
  D --> F
```
After deleting `C`, it becomes
```mermaid
graph TD
  A --> B

  D --> E
  D --> F
```
In this case, `D` is tracked in `danglings` and will be removed but `E` and `F` are still there and not removed.

Reply:
Refined the delete function to delete a dangling tree recursively.

Reply:
Changed the way GC works to address this scenario. Now we traverse `index.json` directly.